### ISSUE TITLE
Method redefinition on 0.7.0 release for "iterate"

### ISSUE THREAD

**Issue by Datseris** [JuliaDiffEq/DiffEqBase.jl#131]
```
Precompiling DynamicalSystems
[ Info: Recompiling stale cache file C:\Users\datseris\.julia\compiled\v0.7\DynamicalSystems\TY1vN.ji for DynamicalSystems [61744808-ddfa-5f27-97ff-6e42cc95d634]
WARNING: Method definition iterate(Any) in module Base at essentials.jl:853 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any, Any) in module Base at essentials.jl:847 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any) in module Base at essentials.jl:853 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any, Any) in module Base at essentials.jl:847 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any) in module Base at essentials.jl:853 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any, Any) in module Base at essentials.jl:847 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any) in module Base at essentials.jl:853 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any, Any) in module Base at essentials.jl:847 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any) in module Base at essentials.jl:853 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any, Any) in module Base at essentials.jl:847 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any) in module Base at essentials.jl:853 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any, Any) in module Base at essentials.jl:847 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\uu9hR\src\integrator_interface.jl:220.
┌ Warning: `cat(dims, As...)` is deprecated, use `cat(As..., dims=dims)` instead.
```
This is after installing/updating on a brand new Julia 0.7.0 installation.

**Comment by ChrisRackauckas** [comment]
We already tagged this

**Comment by Datseris** [comment]
But I just checked out master branch and got the same:
```
(v0.7) pkg> add DiffEqBase#master
  Updating registry at `C:\Users\datseris\.julia\registries\General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
   Cloning git-repo `https://github.com/JuliaDiffEq/DiffEqBase.jl.git`
  Updating git-repo `https://github.com/JuliaDiffEq/DiffEqBase.jl.git`
 Resolving package versions...
┌ Warning: julia version requirement for package DiffEqBase not satisfied
└ @ Pkg.Operations C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v0.7\Pkg\src\Operations.jl:173
  Updating `C:\Users\datseris\.julia\environments\v0.7\Project.toml`
  [2b5f629d] ↑ DiffEqBase v4.21.0 ⇒ v4.21.0+ #master (https://github.com/JuliaDiffEq/DiffEqBase.jl.git)
  Updating `C:\Users\datseris\.julia\environments\v0.7\Manifest.toml`
  [2b5f629d] ↑ DiffEqBase v4.21.0 ⇒ v4.21.0+ #master (https://github.com/JuliaDiffEq/DiffEqBase.jl.git)
  Building DiffEqBase → `C:\Users\datseris\.julia\packages\DiffEqBase\bsDWg\deps\build.log`

julia> using DiffEqBase
[ Info: Recompiling stale cache file C:\Users\datseris\.julia\compiled\v0.7\DiffEqBase\oaevG.ji for DiffEqBase [2b5f629d-d688-5b77-993f-72d75c75574e]
WARNING: Method definition iterate(Any) in module Base at essentials.jl:853 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\bsDWg\src\integrator_interface.jl:220.
WARNING: Method definition iterate(Any, Any) in module Base at essentials.jl:847 overwritten in module DiffEqBase at C:\Users\datseris\.julia\packages\DiffEqBase\bsDWg\src\integrator_interface.jl:220.
```

**Comment by ChrisRackauckas** [comment]
@YingboMa

**Comment by Datseris** [comment]
I also saw the REQUIRE went onto 1.0, so I don't have the actually most recent possible master. But this means that fixes don't propagate into 0.7. I assume some users (some of them being me) would want to stay on 0.7 before they are sure all their stuff can be progressed to 1.0

**Comment by jebej** [comment]
I'm also seeing the same prob on 0.7.
